+NeuroAnalysis/+IO/Export.m

```diff
@@ -10,39 +10,43 @@
     return;
 end
 %% Get dataset and export path
-result.status = false;
-result.source = '';
 if isa(dataset,'char')
-    result.source = dataset;
+    filename = dataset;
     if ~strcmp(sourceformat,'Unknown')
         dataset = NeuroAnalysis.Base.EvalFun(['NeuroAnalysis.',sourceformat,'.Prepare'],[{dataset},varargin]);
         if ~isempty(dataset) && isfield(dataset,'status') && ~dataset.status
+            result = dataset;
+            result.source = NeuroAnalysis.Base.filenamenodirext(filename);
             return;
         end
         filename = NeuroAnalysis.Base.filenamenodirext(dataset.source);
-        exportpath = fullfile(exportpath,[filename,'.mat']);
+        filepath = fullfile(exportpath,[filename,'.mat']);
     else
         [~,filename,ext]=fileparts(dataset);
-        exportpath=fullfile(exportpath,[filename,ext]);
+        filepath=fullfile(exportpath,[filename,ext]);
     end
 end
 %% Save dataset
-disp(['Exporting Dataset:    ',exportpath,'    ...']);
+disp(['Exporting Dataset:    ',filepath,'    ...']);
 if ~strcmp(sourceformat,'Unknown')
-    dataset.filepath = exportpath;
-    save(exportpath,'dataset','-v7.3');
+    dataset.filepath = filepath;
+    save(filepath,'dataset','-v7.3');
 else
-    copyfile(dataset,exportpath);
+    copyfile(dataset,filepath);
+    dataset = struct;
+    dataset.filepath = filepath;
 end
-result.status = true;
-result.source = filename;
 disp('Exporting Dataset:    Done.');
```

Comment:
i don't quite understand all the code change above this line. the result of Export function is mostly exception information. result.status indicate whether successfully prepare the dataset. the result.source is the full path of the file to be prepared. The Dataset, however is the prepared dataset which contain all the information needed to fully describe the dataset. dataset.filepath is the file path of the exported dataset mat file. so after the data exported, the result should be ready containing exception information about the exporting process.